backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -691,6 +682,137 @@ public void validateDiffMergeRequest(
+    final String rootSha = setBlobs(writeSession, blobContainerList, new FileHasher());
+    long timeCreated = new Date().getTime();
+
+    UserInfo currentLoginUserInfo = authService.getCurrentLoginUserInfo();
+    String author = authService.getVertaIdFromUserInfo(currentLoginUserInfo);
+    final String commitSha =
+        VersioningUtils.generateCommitSHA(parentSHAs, commitMessage, timeCreated, author, rootSha);
+
+    Commit internalCommit =
+        Commit.newBuilder()
+            .setDateCreated(timeCreated)
+            .setAuthor(author)
+            .setMessage(commitMessage)
+            .setCommitSha(commitSha)
+            .build();
+    return new CommitEntity(repositoryEntity, parentCommits, internalCommit, rootSha);
+  }
+
+  @Override
+  public RevertRepositoryCommitsRequest.Response revertCommit(
+      RepositoryDAO repositoryDAO, RevertRepositoryCommitsRequest request)
+      throws ModelDBException, NoSuchAlgorithmException {
+    // validating request
+    LOGGER.debug("Validating RevertRepositoryCommitsRequest");
+    if (request.getCommitToRevertSha().isEmpty()) {
+      throw new ModelDBException(
+          "Revert Commit SHA not found in the request", Status.Code.INVALID_ARGUMENT);
+    }
+    if (request.getBaseCommitSha().isEmpty()) {
+      throw new ModelDBException(
+          "Revert Base Commit SHA not found in the request", Status.Code.INVALID_ARGUMENT);
+    }
+    LOGGER.debug("RevertRepositoryCommitsRequest validated");
+
+    List<BlobContainer> blobContainerList;
+    RepositoryEntity repositoryEntity;
+    CommitEntity baseCommitEntity;
+    CommitEntity commitToRevertShaEntity;
+    try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
+      repositoryEntity = repositoryDAO.getRepositoryById(session, request.getRepositoryId());
+
+      if (!VersioningUtils.commitRepositoryMappingExists(
+          session, request.getCommitToRevertSha(), repositoryEntity.getId())) {
+        throw new ModelDBException(
+            "No such revert commit found in the repository : " + request.getCommitToRevertSha(),
+            Status.Code.NOT_FOUND);
+      }
+
+      if (!VersioningUtils.commitRepositoryMappingExists(
+          session, request.getBaseCommitSha(), repositoryEntity.getId())) {
+        throw new ModelDBException(
+            "No such base commit found in the repository : " + request.getCommitToRevertSha(),
+            Status.Code.NOT_FOUND);
+      }
+
+      commitToRevertShaEntity = session.get(CommitEntity.class, request.getCommitToRevertSha());
+      baseCommitEntity = session.get(CommitEntity.class, request.getBaseCommitSha());
+
+      if (commitToRevertShaEntity.getParent_commits() == null
+          || commitToRevertShaEntity.getParent_commits().isEmpty()) {
+        throw new ModelDBException(
+            "No parent found for commit : " + request.getCommitToRevertSha(),
+            Status.Code.NOT_FOUND);
```

Comment:
Code.INTERNAL this is an issue with our system since the parent is not an input parameter

Reply:
Changed